examples/js/loaders/GLTFLoader.js

```diff
@@ -1337,7 +1337,8 @@ THREE.GLTFLoader = ( function () {
 		// BufferGeometry caching
 		this.primitiveCache = {};
 
-		this.textureLoader = new THREE.TextureLoader( this.options.manager );
+		var isWorker = typeof WorkerGlobalScope !== 'undefined' && self instanceof WorkerGlobalScope;
+		this.textureLoader = isWorker ? new THREE.ImageBitmapLoader( this.options.manager ) : new THREE.TextureLoader( this.options.manager );
```

Comment:
Is this working for you? I think I'd expect it to work in Chrome but not anywhere else...

Reply:
It is working in Chrome and IE 11. I took it from https://gist.github.com/miguelmota/a0c9aa9398507e606a6e
https://stackoverflow.com/questions/7507638/any-standard-mechanism-for-detecting-if-a-javascript-is-executing-as-a-webworker
https://stackoverflow.com/questions/7931182/reliably-detect-if-the-script-is-executing-in-a-web-worker/18002694

Reply:
The detection looks OK, but I thought ImageBitmap didn't work in IE11? https://developer.mozilla.org/en-US/docs/Web/API/ImageBitmap

[This demo](https://threejs.org/examples/#webgl_loader_imagebitmap) is working in Firefox and Safari though, so maybe I'm just confused.

Reply:
I don't think I follow the recent glTF loader update... but can't user specify image loader type (`ImageLoader` or `ImageBitmapLoader`) used in `TextureLoader` yet via `LoadingManager` or somewhere else?

Using `ImageBitmap` still has some difficulties due to different API from regular images, for example https://github.com/mrdoob/three.js/issues/16144, so I think it would be safer that user who knows the `ImageBitmap` limitation specifies image loader type in user code side if possible. And we'll have less code change.

Reply:
I have had more tests with different models and found that this line has to be inside of TextureLoader, because of TextureLoader is producing a Texture(I have missed it).

Regarding IE 11 it was just a test of that check will it work correctly, but in the end, it is up to a client to prevent loading GLTFLoader inside WebWorker for IE11.

Reply:
> I don't think I follow the recent glTF loader update... but can't user specify image loader type (`ImageLoader` or `ImageBitmapLoader`) used in `TextureLoader` via `LoadingManager` or somewhere else yet?
> 
> Using `ImageBitmap` still has some difficulties due to different API from regular images, for example #16144, so I think it would be safer that user who knows the `ImageBitmap` limitation specifies image loader type in user code side if possible. And we'll have less code change.

Without this changes, TextureLoader simply does not work inside WebWorker, so you are limited to OffscreenCanvas https://caniuse.com/#search=OffscreenCanvas.

But yes it will be great if we could define ImageBitmapLoader.options for a TextureLoader.
Maybe we can add some temporary solution static default options
`
ImageBitmapLoader.defaultOptions = undefined;

function ImageBitmapLoader( manager ) {
        //....
	this.options = ImageBitmapLoader.defaultOptions;
}
`